docs/testing.md

```diff
@@ -0,0 +1,140 @@
+## Testing
+
+`tsaga` comes with its testing utilities.
+
+The utilities are similar to `redux-saga-test-plan` and modeled after how a saga would run in your application.
+
+### Sagas
+
+The assertions use the `expect` function from the `jest` testing framework.
+
+```typescript
+import { expectSaga, select, call, run, spawn } from 'tsaga';
+
+it('should run a saga', () => {
+  return (
+    expectSaga(watchForFetchUser)
+      // Setup a store with a reducer and an optional initial state
+      // This is optional
+      .withReducer(rootReducer, undefined)
+      // Setup some mocks which should be used instead of the real functions
+      // The mock helper functions provide useful type assertions,
+      // so that you return a value which is valid
+      // This is optional
+      .withMocks([
+        // A select mock, the second argument is the return value when this mock is used
+        select(getUserById, { id: 5 }),
+        // A call mock, the second argument is the return value when this mock is used
+        call(fetchUserApi, { id: 5 }),
+        // A spawn mock, the second argument is the return value when this mock is used
+        spawn(trackUser),
+        // A run mock, the second argument is the return value when this mock is used
+        run(loginUser),
+      ])
+      // These are the expectations for the saga
+      // Expect the saga to call a function
+      // The arguments afterwards are what will be passed to the function
```

Comment:
`will be passed` -> `are expected to be passed` as the assertions don't call anything themselves.